### PR TITLE
2583 table head improve unit test coverage

### LINKED ISSUE #2583: [TableHead] improve unit test coverage

### PR DESCRIPTION
**Pull request by bjornalm**
Closes #2583

**Summary**
Improved testing coverage for table head

**Change List (commits, features, bugs, etc)**
Added unit tests

**Acceptance Test (how to verify the PR)**
test coverage is 90% or above

**Regression Test (how to make sure this PR doesn't break old functionality)**

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/Table/TableHead/TableHead.test.jsx [hunk 1]
```diff
@@ -1,10 +1,11 @@
 import React from 'react';
 import { mount } from 'enzyme';
```
Comment (kevinsperrine) [inline packages/react/src/components/Table/TableHead/TableHead.test.jsx:2]: Nice work! Do we want to remove enzyme this go around or open an issue to tackle it later?

**Review by kevinsperrine — APPROVED** [review]
Looks great! Just the enzyme question.

File: packages/react/src/components/Table/TableHead/TableHead.test.jsx (hunk 1 above)
Comment (bjornalm) [reply packages/react/src/components/Table/TableHead/TableHead.test.jsx:2]: Good question. There are a lot of enzyme tests for this component. I think we should do that in a separate move, the primary objective was the coverage increase.

Comment (bjornalm) [reply packages/react/src/components/Table/TableHead/TableHead.test.jsx:2]: I tend to rewrite the tests when it feels feasible in relation to the effort of increasing the coverage

Comment (kevinsperrine) [reply packages/react/src/components/Table/TableHead/TableHead.test.jsx:2]: agreed. Just wanted to check.

**Review by davidicus — APPROVED** [review]

**Pull request merged**